### DISCUSSION TITLE
How to add callbackUrl for signin/signout for preview deployments?

### DISCUSSION THREAD

**Discussion by jcheese1** [nextauthjs/next-auth#556] [Help]
*Please refer to the [documentation](https://next-auth.js.org/getting-started/introduction), the [example project](https://github.com/iaincollins/next-auth-example) and existing issues before creating a new issue.*

**Your question**
A clear and concise question.

I guess this is more on how to obtain the baseUrl in Next.js, but searched around to no avail. How can I setup an absolute url in the frontend for callbackUrl in preview deployments? Is it possible?

Thanks

**What are you trying to do**
A description of what you are trying to do.

Trying to setup a callbackUrl in preview deployments

**Documentation feedback**
*Documentation refers to searching through [online documentation](https://next-auth.js.org), code comments and issue history. The example project refers to [next-auth-example](https://github.com/iaincollins/next-auth-example).*

* [x] Found the documentation helpful
* [ ] Found documentation but was incomplete
* [ ] Could not find relevant documentation
* [ ] Found the example project helpful
* [ ] Did not find the example project helpful

**Comment by camsloanftc** [comment]
I think if you're deploying to Vercel, you're probably going to need something like `VERCEL_URL` from this list of system environment variables:

https://vercel.com/docs/v2/build-step#system-environment-variables

I actually haven't deployed anything this way yet, so I could be wrong, but that's where I would start.
